.github/workflows/build-and-test-linux.yaml

```diff
@@ -30,14 +30,30 @@ jobs:
         with:
           submodules: "recursive"
       - name: Build
-        run: sudo su -c "PG_VERSION=$PG_VERSION USE_SOURCE=1 ./ci/scripts/build-linux.sh"
+        id: build
+        run: sudo su -c "PG_VERSION=$PG_VERSION USE_SOURCE=1 GITHUB_OUTPUT=$GITHUB_OUTPUT ./ci/scripts/build-linux.sh"
         env:
           PG_VERSION: ${{ matrix.postgres }}
           BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+      - name: Create Archive Package
+        id: archive
+        run: sudo su -c "GITHUB_OUTPUT=$GITHUB_OUTPUT ./ci/scripts/package-archive.sh"
+        if: ${{ matrix.postgres == '15' }} # The output is not dependent to pg version, so we can run this only once
       # Enable tmate debugging of manually-triggered workflows if the input option was provided
       # - name: Setup tmate session
       #   uses: mxschmitt/action-tmate@v3
       - name: Run tests
         run: sudo su postgres -c "PG_VERSION=$PG_VERSION ./ci/scripts/run-tests.sh"
         env:
           PG_VERSION: ${{ matrix.postgres }}
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
```

Comment:
Will these run when someone forks the repo? doe these use any secrets?
We should properly gate any workflows that use secrets so people do not get annoying build fail emails after forking our repository